### PR TITLE
Add mergeLeaves function and make matchSeriesEnum stricter

### PR DESCRIPTION
**Pull request by kwahlin**
As per new requirements

### CODE REVIEW

File: whelktool/scripts/cleanups/2021/04/lxl-1931-merge-series-membership.groovy [hunk 1]
```diff
@@ -56,8 +56,8 @@ selectBySqlWhere(where) { data ->
         Map a = cmpData[it[0]]
         Map b = cmpData[it[1]]
 
-        Map mergedTrees = merge(a.tree, b.tree)
-        Map leaves = a.leavesForCmp + b.leavesForCmp
+        Map mergedLeaves = mergeLeaves(a.leavesForCmp, b.leavesForCmp, b.tree.inSeries?.instanceOf.asBoolean())
```
Comment (olovy) [inline whelktool/scripts/cleanups/2021/04/lxl-1931-merge-series-membership.groovy:59]:
```suggestion
        Map mergedLeaves = mergeLeaves(a.leavesForCmp, b.leavesForCmp, b.tree.inSeries?.instanceOf)
```
inSeries == null is already coerced to false
asBoolean() will lead to nullpointer if instanceOf is missing?
?

**Review by olovy — APPROVED** [review]
LGTM!
see comment

File: whelktool/scripts/cleanups/2021/04/lxl-1931-merge-series-membership.groovy (hunk 1 above)
Comment (kwahlin) [reply whelktool/scripts/cleanups/2021/04/lxl-1931-merge-series-membership.groovy:59]: @olovy actually null.asBoolean() evaluates to false so should be fine!

Comment (olovy) [reply whelktool/scripts/cleanups/2021/04/lxl-1931-merge-series-membership.groovy:59]: Ok! TIL LOL

**Pull request merged**